### PR TITLE
Oxli prototype using fastq-to-fasta, abund-dist-single, {build/load}-graph

### PR DESCRIPTION
**Pull request by bocajnotnef**
- [ ] Is it mergable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment. 
- [x] Is it documented in the ChangeLog?
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by bocajnotnef** [comment]
retest this please

File: scripts/abundance-dist-single.py [hunk 1]
```diff
@@ -68,116 +46,16 @@ def main():  # pylint: disable=too-many-locals,too-many-branches
...
-        print >> hist_fp, _, i, sofar, round(frac, 3)
-
-        if sofar == total:
-            break
-
-    if args.savetable:
-        print >>sys.stderr, 'Saving k-mer counting table ', args.savetable
-        print >>sys.stderr, '...saving to', args.savetable
-        counting_hash.save(args.savetable)
+    abund_dist_single.do_abund_dist_single(args.input_sequence_filename,
```
Comment (bocajnotnef) [inline scripts/abundance-dist-single.py]: @mr-c I feel like to refactor this into something simpler I'd just end up re-inventing argparse, just in a function-oriented way.

**Comment by bocajnotnef** [comment]
retest this please

File: scripts/abundance-dist-single.py (hunk 1 above)
Comment (mr-c) [inline scripts/abundance-dist-single.py]: Break up abund_dist_single into reusable components.

What is the core of the script? Creating an abundance histogram from a counting table object and ReadParser object.

Create a python method that does that and returns the result. Move this method into Oxli.

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
@mr-c @brtaylor92 @b-wyss @jessicamizzi @wrightmhw CR please?

**Comment by mr-c** [comment]
@brtaylor92 @b-wyss @jessicamizzi @wrightmhw: no worries, I've got this

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
- [ ] Apply object-oriented philosophy to improve usability in relation to ipython like tools; @mr-c to elaborate

**Comment by mr-c** [comment]
- [ ] argparse only in oxli
- [ ] central functionality of each script as a python method with object
inputs and outputs (not filenames)

**Comment by mr-c** [comment]
- [ ] argparse only in oxli
 - [ ] central functionality of each script as a python method with object
inputs and outputs (not filenames)

**Comment by bocajnotnef** [comment]
@mr-c So am I gonna try to nail this stuff down prior to a merge?

File: ChangeLog [hunk 2]
```diff
@@ -19,6 +28,7 @@
 
     * README.rst,doc/user/install.txt: clarify that we support Python 2.7.x
     and not Python 3.
+>>>>>>> d4621d3831fa724f975c1acb323d6aece08dfbc4
```
Comment (mr-c) [inline ChangeLog]: :-)

Comment (bocajnotnef) [inline ChangeLog]: Siiiigggghhhh...

**Comment by bocajnotnef** [comment]
Code smells.....?

**Comment by mr-c** [comment]
My apologies; I don't know what Landscape.io brought itself back from the
dead. I've revoked its access.

**Comment by bocajnotnef** [comment]
Currently unsure what to do with oxli; could attempt to bring it back up to sync with the rest of khmer but I feel like the merge conflicts would be brutal. Now that I have a better idea of what I'm doing (refactoring in addition to migrating the scripts to importable funcs) I could restart oxli and actually break the scripts down into usable functions. 

@mr-c @ctb suggestions?

**Comment by mr-c** [comment]
A new PR is fine. Bring over whatever you feel is useful.

Let's get screed 0.8 out first!

**Comment by bocajnotnef** [comment]
Agreed; @ctb requested a refresh on the status of all pulls/issues I'm involved in. Hence, updating.

**Comment by ctb** [comment]
Would suggest starting new PR with some very simple scripts - split-paired-reads, extract-paired-reads, and interleave-reads.

**Comment by mr-c** [comment]
Superseded by #961

**Pull request closed without merging**